tests/azure_iot_hub_client_ut.c

```diff
@@ -0,0 +1,351 @@
+
+    /* Fail when smaller buffer is passed */
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  ucHostname, sizeof( ucHostname ) - 1,
+                                                  ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                                  &xHubClientOptions,
+                                                  ucBuffer, azureiotUSERNAME_MAX,
+                                                  prvGetUnixTime, &xTransportInterface ),
+                          eAzureIoTHubClientSuccess );
+
+    /* Fail init when AzureIoTMQTT_Init fails */
+    will_return( AzureIoTMQTT_Init, eAzureIoTMQTTNoMemory );
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  ucHostname, sizeof( ucHostname ) - 1,
+                                                  ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                                  &xHubClientOptions,
+                                                  ucBuffer, sizeof( ucBuffer ),
+                                                  prvGetUnixTime, &xTransportInterface ),
+                          eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Init_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+    AzureIoTHubClientOptions_t xHubClientOptions = { 0 };
+
+    ( void ) state;
+    will_return( AzureIoTMQTT_Init, eAzureIoTMQTTSuccess );
+
+    assert_int_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                              ucHostname, sizeof( ucHostname ) - 1,
+                                              ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                              &xHubClientOptions,
+                                              ucBuffer,
+                                              sizeof( ucBuffer ),
+                                              prvGetUnixTime,
+                                              &xTransportInterface ),
+                      eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Deinit_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    AzureIoTHubClient_Deinit( &xTestIoTHubClient );
+}
+
+static void testAzureIoTHubClient_Connect_InvalidArgFailure( void ** state )
+{
+    ( void ) state;
+
+    bool xSessionPresent;
+
+    assert_int_not_equal( AzureIoTHubClient_Connect( NULL,
+                                                     false,
+                                                     &xSessionPresent,
+                                                     60 ),
+                          eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Connect_MQTTConnectFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    bool xSessionPresent;
```

Comment:
>xSessionPresent [](start = 9, length = 15)

nit: start of function

Reply:
all the places

---
In reply to: [612735034](https://github.com/Azure/azure-iot-middleware-freertos/pull/83#discussion_r612735034) [](ancestors = 612735034)